### ISSUE TITLE
[clang-tidy] performance-move-const-arg warning should not be triggered when passing to function taking a rvalue reference

### ISSUE THREAD

**Issue by fabiorossetto** [llvm/llvm-project#52746]
Given the following snippet of code:
```c++
#include <utility>

void func(int&& a) {}

void use_func() {
    int a;
    func(std::move(a));
}
```
`clang-tidy` complains as follows:
```
<source>:9:10: warning: std::move of the variable 'a' of the trivially-copyable type 'int' has no effect; remove std::move() [performance-move-const-arg]
    func(std::move(a));
```
however removing `std::move` as suggested causes the code to not compile:
```
<source>:9:5: error: no matching function for call to 'func' [clang-diagnostic-error]
    func(a);
    ^~~~
<source>:3:6: note: candidate function not viable: expects an rvalue for 1st argument
void func(int&& a) {
     ^
```
I would argue that if `std::move` is used when passing a variable to a function accepting rvalue reference arguments, then the use of `std::move` is justified and should not be flagged, even if the type is trivially copyable.

**Comment by carlosgalvezp** [comment]
Duplicate of #43359
